include/fs_utils.h

```diff
@@ -168,12 +199,31 @@ std::deque<std_fs::path> get_xdg_data_dirs() noexcept;
 
 union FatAttributeFlags; // forward declaration
 
-FILE* local_drive_create_file(const std_fs::path& path,
-                              const FatAttributeFlags attributes);
+NativeFileHandle local_drive_open_file(const std_fs::path& path,
+                                       const bool write_access);
+NativeFileHandle local_drive_create_file(const std_fs::path& path,
+                                         const FatAttributeFlags attributes);
 uint16_t local_drive_create_dir(const std_fs::path& path);
 uint16_t local_drive_get_attributes(const std_fs::path& path,
                                     FatAttributeFlags& attributes);
 uint16_t local_drive_set_attributes(const std_fs::path& path,
                                     const FatAttributeFlags attributes);
 
+// Windows ReadFile and WriteFile take in a DWORD (32 bit unsigned) for number of bytes
+// We'll make that our limit until someone can give a good reason why they need to read/write more than 4GB (unlikely)
+NativeFileReturn local_drive_read_file(const NativeFileHandle handle, uint8_t *buffer, const uint32_t bytes_requested);
+NativeFileReturn local_drive_write_file(const NativeFileHandle handle, const uint8_t *buffer, const uint32_t bytes_requested);
```

Comment:
I guess we could even just use `int` for all these byte sizes because of the FAT16 limits if it's easier overall.

Reply:
I suppose I can go up to `int64_t`.  I would need to write some additional code to break up large (greater than 4GB) reads/writes on Windows into multiple calls but that'll be easy enough.

Reply:
Changed to `int64_t`